Makefile

```diff
@@ -1,25 +1,183 @@
+
+# This is only used to help during local opam package
+# development
+opam: build
+	$(TRV) opam make-opam --target-dir $(CURDIR) --name $(NAME) \
+	--semver $(SEMVER) --homepage $(HOMEPAGE) --dev-repo $(DEV_REPO) \
+	--lib-dir $(LIB_DIR) --license $(LICENSE) --author $(AUTHOR) \
+	--maintainer $(AUTHOR) --bug-reports $(BUG_REPORTS) \
+	--build-cmd "make" --install-cmd 'make "install" "PREFIX=%{prefix}%" \
+	"SEMVER=%{vrt:version}%"' --remove-cmd 'make "remove" \
+	"PREFIX=%{prefix}%"' $(BUILD_MOD_DEPS) $(MOD_DEPS)
+
+unpin-repo:
+	opam pin remove -y $(NAME)
+
+pin-repo:
+	opam pin add -y $(NAME) $(CURDIR)
+
+install-local-opam: opam pin-repo
+	opam remove $(NAME); opam install $(NAME)
+
+prepare: build
+	$(TRV) opam prepare --organization $(ORGANIZATION) \
+	--target-dir $(BUILD_DIR) --homepage $(HOMEPAGE) \
+	--dev-repo $(DEV_REPO) --lib-dir $(LIB_DIR) --license $(LICENSE) \
+	--name $(NAME) --author $(AUTHOR) --maintainer $(AUTHOR) \
+	--bug-reports $(BUG_REPORTS) --build-cmd "make" \
+	--install-cmd 'make "install" "PREFIX=%{prefix}%"' \
+	--remove-cmd 'make "remove" "PREFIX=%{prefix}%"' $(BUILD_MOD_DEPS) \
+	$(MOD_DEPS) --description-file '$(DESC_FILE)'
+
+install-extra: build
+	mkdir $(PREFIX)/bin
+	cp $(BUILD_DIR)/lib/vrt.native $(PREFIX)/bin/vrt
+
+install-library: metadata
+	cd $(LIB_DIR); ocamlfind install $(NAME) META `find ./  -name "*.cmi" \
+	-o -name "*.cmo" -o -name "*.o" -o -name "*.cmx" -o -name "*.cmxa" \
+	-o -name "*.cmxs" -o -name "*.a" -o -name "*.cma"`
+
+install: install-library install-extra
+
+submit: prepare
+	opam-publish submit $(BUILD_DIR)/$(NAME).$(SEMVER)
+
+remove:
+	ocamlfind remove $(NAME)
+
+clean:
+	rm -rf $(CLEAN_TARGETS)
+	rm -rf $(BUILD_DIR)
+
+# =============================================================================
+# Rules for testing
+# =============================================================================
+
+compile-tests: $(TEST_RUN_CMDS)
+
+$(TEST_RUN_CMDS): $(TESTS_DIR)/.d
+	$(BUILD) $(notdir $@).byte
+	@find $(LIB_DIR) -name $(notdir $@).byte -exec cp {} $(@) \;
+
+$(TEST_RUN_TARGETS): run-%: $(TESTS_DIR)/%
+	$<
+
+test: build unit-test integ-test
+
+unit-test: $(filter %_unit_tests_run, $(TEST_RUN_TARGETS))
+
+integ-test: $(filter %_integ_tests_run, $(TEST_RUN_TARGETS))
+
+# =============================================================================
+# Support
+# =============================================================================
+$(UTOP_INIT): build
+	@echo "$(UTOP_MODS)" > $(UTOP_INIT)
+	@echo "open Core.Std;;" >> $(UTOP_INIT)
+	@echo "open Async.Std;;" >> $(UTOP_INIT)
+	@echo '#load "$(NAME).cma";;' >> $(UTOP_INIT)
 
-install-extra:
-	cp $(BUILD_DIR)/lib/trv.native $(PREFIX)/bin/trv
+utop: $(UTOP_INIT)
+	utop -I $(LIB_DIR) -init $(UTOP_INIT)
 
-trv.mk:
-	trv build gen-mk
+.merlin: build
+	$(TRV) build make-dot-merlin \
+		--build-dir $(BUILD_DIR) \
+		--lib "$(DEPS)" \
+		--source-dir $(SOURCE_DIR)
 
--include trv.mk
+shell:
+	nix-shell -p $(SHELL_DEPS)
```

Comment:
Yeah, I just wanted to make sure this was documented somewhere if it isn't yet, since I hadn't seen it yet, agree on that being in our centralized docs.